.github/workflows/test.yml

```diff
@@ -52,10 +61,29 @@ jobs:
         -DOPENASSETIO_ENABLE_SANITIZER_ADDRESS=ON 
         -DOPENASSETIO_ENABLE_SANITIZER_UNDEFINED_BEHAVIOR=ON
 
-    - name: Build, install and test
+    - name: Configure CMake build (Windows)
+      if: runner.os == 'Windows'
+      run: >
+        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
```

Comment:
This actually isn't great since the path depends on the runner version. This will be different for a new Windows/compiler version. I saw some people managed to store it inside the matrix but I don't know YAML well enough right now to fix it straight away.

Reply:
The matrix stuff and the reference to it, e.g. `${{ matrix.os }}`, isn't YAML syntax, it's Github Action specific, so I would refer you to their docs.

However, there is also a way in YAML to avoid repetition - anchors and aliases, e.g. see https://www.educative.io/blog/advanced-yaml-syntax-cheatsheet

Reply:
Perhaps under the matrix we can add a new variable under `matrix:`, e.g.

```
console_preamble: 
  # Ubuntu
  - ""
  # Windows
  - call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
```
Then add a `${{ matrix.console_preamble }}` line to the `run` blocks.

Does seem a bit of a waste having to have a blank Ubuntu line.  Surely there's a way in Github Actions to set a common variable outside the matrix, but I haven't looked.

You could perhaps add even more matrix parameters to parametrise the cmake command-line differences, avoiding the need for `if: runner.os`.  But that might not be possible, or be more hassle than it's worth.

OTIO's Github Actions may be a good reference.

Reply:
Btw - it's totally fine if the windows setup is a bit "hard coded" at this point, we can't sink too much time into it at this point. Just having a working version is good for now, even if it breaks in a month or so.